django_cloud_deploy/config.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""A module about configuration files generation."""
+
+import os
+from typing import Dict, List, Optional
+import yaml
+
+
+class Configuration(object):
+    """A class to encapsulate a single configuration."""
+
+    _HEADER = '# Generated file, do not edit'
+
+    def __init__(self, django_directory_path: str):
+        if not os.path.isdir(django_directory_path):
+            raise ValueError('[{}] is not a valid directory path.'.format(
+                django_directory_path))
+        self._config_path = os.path.join(django_directory_path,
+                                         '.config.yaml')
+        if os.path.exists(self._config_path):
+            with open(self._config_path) as config_file:
+                self._data = yaml.load(config_file)
+
+        else:
+            self._data = {}
+
+    @staticmethod
+    def _generate_lines(data: Dict[str, str]) -> List[str]:
+        """Generate text lines in yaml format from the given dictionary.
+
+        For example, given data = '{'key1': 'value1', 'key2': 'value2'}', the
+        output is: ['key1: value1', 'key2: value2']
+
+        Args:
+            data: The dictionary we want to use to generate lines.
+
+        Returns:
+            A list of text in yaml format.
+        """
+        return ['{}: {}'.format(k, v) for k, v in data.items()]
+
+    def set(self, attr: str, value: str):
+        self._data[attr] = value
+
+    def save(self):
+        """Generate the configuration file in yaml format."""
+        lines = [self._HEADER] + self._generate_lines(self._data)
+        yaml_text = '\n'.join(lines)
+        with open(self._config_path, 'w') as config_file:
+            config_file.write(yaml_text)
+
+    def get(self, attr: str) -> Optional[str]:
+        """Get value of the specified attribute.
+
+        Args:
+            attr: The attribute name we want to get value.
+
+        Returns:
+            Value of the specified attribute if it exist. Otherwise return None.
+        """
+        try:
+            return self._data[attr]
```

Comment:
return self._data.get(attr) and you don't need the try block

Reply:
Done